### PR TITLE
[SYCL] Try to fix MacOS build after #11956

### PR DESCRIPTION
**Pull request by aelovikov-intel**
MacOS' STL is not standard-conformant, try to workaround for that in our code.

### CODE REVIEW

File: .github/workflows/sycl_post_commit.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ on:
     - ./devops/actions/cleanup
     - ./devops/actions/cached_checkout
 
+
```
Comment (aelovikov-intel) [inline .github/workflows/sycl_post_commit.yml:21]:
```suggestion
```

File: sycl/source/builtins/relational_functions.cpp [hunk 2]
```diff
@@ -16,6 +16,15 @@
 namespace sycl {
 inline namespace _V1 {
 
+template <typename T> static auto process_arg_for_macos(T x) {
```
Comment (cperkinsintel) [inline sycl/source/builtins/relational_functions.cpp:19]: should this function be declared constexpr?  I think it's not strictly required, but if that's its sole intended use, might be best.

**Review by cperkinsintel — APPROVED** [review]
Had one question, but otherwise looks good.

File: sycl/source/builtins/relational_functions.cpp (hunk 2 above)
Comment (aelovikov-intel) [reply sycl/source/builtins/relational_functions.cpp:19]: `x` is a run-time value most of the time...

**Pull request merged**